libs/utils/android/system.py

```diff
@@ -45,12 +46,24 @@ def systrace_start(env, trace_file, time=None,
         will keep tracing until systrace_stop() is called if no value is passed
         :type time: int
 
-        :param events: events to trace
-        :type events: str list
+        :param conf: custom systrace configuration
+        will be used instead of env.conf['systrace'] if set
+        :type conf: dict
         """
 
         log = logging.getLogger('System')
 
+        # Prioritize custom conf, then environment conf, then default conf
+        if conf is not None:
+            conf = conf
+        elif 'systrace' in env.conf and conf is None:
+            conf = env.conf['systrace']
+        else:
+            conf = {}
```

Comment:
Yes, that should work too. I liked the if/elif/else though because it's pretty clear what are the priorities (at least to me). On a sidenote, there's a refactoring derp here because `elif 'systrace' in env.conf and conf is None:` should just be `elif 'systrace in env.conf`.